fdbserver/Coordination.actor.cpp

```diff
@@ -563,37 +578,83 @@ ACTOR Future<Void> leaderServer(LeaderElectionRegInterface interf, OnDemandStore
 			if (forward.present()) {
 				req.reply.send(forward.get());
 			} else {
+				StringRef reqClusterName = getClusterName(req.key);
+				StringRef clusterName = getClusterName(ccf->getConnectionString().clusterKey());
+				if (reqClusterName.compare(clusterName) ||
+					ccf->getConnectionString().coordinators() != req.coordinators) {
+					TraceEvent(SERVER_KNOBS->ERROR_ON_CROSS_CLUSTER_VIOLATION ? SevError : SevWarn, "CCFMismatch")
+					    .detail("RequestType", "ElectionResultRequest")
+					    .detail("LocalCS", ccf->getConnectionString().toString())
+					    .detail("IncomingClusterKey", req.key)
+					    .detail("IncomingCoordinators", describeList(req.coordinators, req.coordinators.size()));
+				}
 				regs.getInterface(req.key, id).electionResult.send(req);
 			}
 		}
 		when(GetLeaderRequest req = waitNext(interf.getLeader.getFuture())) {
 			Optional<LeaderInfo> forward = regs.getForward(req.key);
 			if (forward.present())
 				req.reply.send(forward.get());
-			else
+			else {
+				StringRef reqClusterName = getClusterName(req.key);
+				StringRef clusterName = getClusterName(ccf->getConnectionString().clusterKey());
+				if (reqClusterName.compare(clusterName)) {
```

Comment:
If we only compare the cluster name, instead of coordinators + ID, it's assuming no two clusters would ever have the same name, which is hard to enforce and unrealistic IMO.

Reply:
I'm not sure using the random key is a good way to distinguish clusters with the same name though (based off the discussions I've had!). And an administrator who opts into using this PR's log would know not to create to clusters with the same name.

Reply:
IIRC, the random part changes when there is a change CD command issued from CLI, [here](https://github.com/apple/foundationdb/blob/master/fdbclient/ManagementAPI.actor.cpp#L1091).

CDs run disk paxos for state(`DBCoreState` and forward info) replication and run leader election algorithm for electing `ClusterController`. There isn't a concept of `master` for CDs.

Reply:
yes- thanks for the clarification